drivers/include/at24cxxx.h

```diff
@@ -0,0 +1,218 @@
+/**
+ * @defgroup    drivers_at24cxxx AT24CXXX EEPROM unit
+ * @brief       Device driver interface for the AT24CXXX EEPROM units
+ *
+ * @{
+ *
+ * @file
+ * @brief       Device driver interface for AT24CXXX EEPROM units.
+ *
+ */
+
+#ifndef AT24CXXX_H
+#define AT24CXXX_H
+
+#include <stdint.h>
+
+#include "periph/gpio.h"
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * return values
+ */
+enum {
+    AT24CXXX_OK,
+};
+
+/**
+ * AT24CXXX device tyes
+ */
+typedef enum {
+    AT24CXXX_DEV_TYPE_AT24C128,
+    AT24CXXX_DEV_TYPE_AT24C256
+} at24cxxx_dev_type_t;
+
+/**
+ * 7 bit I2C device address
+ * 10100 A1 A0
+ */
+enum {
+    AT24CXXX_DEV_ADDR_00    = (0x50),   //A1 = 0, A0 = 0
+    AT24CXXX_DEV_ADDR_01    = (0x51),   //A1 = 0, A0 = 1
+    AT24CXXX_DEV_ADDR_10    = (0x52),   //A1 = 1, A0 = 0
+    AT24CXXX_DEV_ADDR_11    = (0x53)    //A1 = 1, A0 = 1
+};
+
+/**
+ * @brief struct that holds initialization parameters
+ */
+typedef struct at24cxxx_params {
+    i2c_t i2c;                      /**< I2C bus number */
+    uint8_t dev_addr;               /**< I2C device address */
+    gpio_t pin_wp;                  /**< write protect pin */
+    at24cxxx_dev_type_t dev_type;   /**< device type */
```

Comment:
I have the feeling that this driver will work for many other I2C EEPROMS too.
And since the type is only used to calculate the size, why not just store the size directly in this struct instead?

Reply:
I thought having functions that evaluate device attributes by a device type would be better, than having a member for every device attribute. But you are right, so far it is just a single attribute, the EEPROM size. If this driver could work with other EEPROMs too, the address length and write cycle delay may also differ. Would it then still be better to have a member for every device specific attribute?

Reply:
Only if it makes sense. We can always add those once it becomes necessary. But let's not make the code needlesly complicated to future proof it for some hypothetical EEPROMs that might not even exist.

If we poll with a delay, we don't need the write cycle delay. I see that 16 bit addresses are a limitation if we want to use EEPROMS > 64kB, but that can be addressed it a separate PR.